unity/Assets/Scripts/ActionDispatcher.cs

```diff
@@ -167,75 +167,74 @@ public static class ActionDispatcher {
 
     private static List<MethodInfo> getCandidateMethods(Type targetType, string action) {
         Dictionary<string, List<MethodInfo>> methodDispatchTable = getMethodDispatchTable(targetType);
-        if (!methodDispatchTable.ContainsKey(action))
-        {
+        if (!methodDispatchTable.ContainsKey(action)) {
             List<MethodInfo> methods = new List<MethodInfo>();
 
             List<Type> hierarchy = new List<Type>();
+
             // not completely generic
             Type ht = targetType;
-            while (ht != typeof(object))
-            {
+            while (ht != typeof(object)) {
                 hierarchy.Add(ht);
                 ht = ht.BaseType;
             }
 
-            foreach (MethodInfo mi in getMethods(targetType))
-            {
-                if (mi.ReturnType != typeof(void) || mi.Name != action)
-                {
+            foreach (MethodInfo mi in getMethods(targetType)) {
+                if (mi.ReturnType != typeof(void) || mi.Name != action) {
                     // We only allow dispatching to public void methods
                     continue;
                 }
                 bool replaced = false;
+
                 // we do this to handle the case of a child method hiding a method in the parent
                 // in which case both methods will show up.  This happens if virtual, override or new
                 // are not used
                 ParameterInfo[] sourceParams = mi.GetParameters();
 
-                for (int j = 0; j < methods.Count && !replaced; j++)
-                {
+                for (int j = 0; j < methods.Count && !replaced; j++) {
                     bool signatureMatch = true;
                     ParameterInfo[] targetParams = methods[j].GetParameters();
                     int minCommon = Math.Min(sourceParams.Length, targetParams.Length);
-                    for (int k = 0; k < minCommon; k++)
-                    {
-                        if (sourceParams[k].ParameterType != targetParams[k].ParameterType)
-                        {
+                    for (int k = 0; k < minCommon; k++) {
+                        if (sourceParams[k].ParameterType != targetParams[k].ParameterType) {
                             signatureMatch = false;
                         }
                     }
 
-                    if (sourceParams.Length > targetParams.Length && !sourceParams[minCommon].HasDefaultValue)
-                    {
-                        signatureMatch = false;
-                    }
-                    else if (targetParams.Length > sourceParams.Length && !targetParams[minCommon].HasDefaultValue)
-                    {
-                        signatureMatch = false;
+                    if (sourceParams.Length != targetParams.Length) {
```

Comment:
Would `signatureMatch && (sourceParams.Length != targetParams.Length)` be more efficient?